packages/rich-list-core/src/saga/AddressParser/dftx/AnyAccountToAccount.ts

```diff
@@ -0,0 +1,16 @@
+import { fromScript } from '@defichain/jellyfish-address'
+import { AnyAccountToAccount, CAnyAccountToAccount, DfTx } from '@defichain/jellyfish-transaction'
+import { DfTxAddressParser } from './_abstract'
+
+export class AnyAccountToAccountParser extends DfTxAddressParser<AnyAccountToAccount> {
+  OP_CODE: number = CAnyAccountToAccount.OP_CODE
+
+  extract (anyAccountToAccount: DfTx<AnyAccountToAccount>): string[] {
+    const fromAddresses = anyAccountToAccount.data.from
+      .map((fromScriptBal) => fromScript(fromScriptBal.script, this.network)?.address as string)
+    const toAddresses = anyAccountToAccount.data.to
+      .map((toScriptBal) => fromScript(toScriptBal.script, this.network)?.address as string)
+
+    return [...fromAddresses, ...toAddresses]
+  }
```

Comment:
```suggestion
  extract (anyAccountToAccount: DfTx<AnyAccountToAccount>): string[] {
    const fromAddresses = this.getScriptBalanceAddresses(anyAccountToAccount.data.from)
    const toAddresses = this.getScriptBalanceAddresses(anyAccountToAccount.data.to)

    return [...fromAddresses, ...toAddresses]
  }

  private getScriptBalanceAddresses (balances: ScriptBalances[]): string[] {
    return balances.reduce<string[]>((addresses, balance) => {
      const address = fromScript(balance.script, this.network)?.address
      if (address !== undefined) {
        addresses.push(address)
      }
      return addresses
    }, [])
  }
```

I think perhaps this suggestion allows us to avoid having to cast a potential `undefined` value to a `string`. Introducing a an address extractor which ignores any `undefined` outcomes of `fromScript`

Reply:
Since this is re-occuring in multiple files perhaps build the address extractor into the `DfTxAddressParser` class as a utility method.

Reply:
Quickly attempted, felt quite unnecessary.
1. They are decoding txs those already on chain, the extracted address should never be undefined, casting is solely for linter.
2. Casted into "undefined" might be not a bad idea and it will catch attention if any bad decoding.

Reply:
Yup, review was mostly based on investigating potential output of `fromScript` which seems to catch and ignore errors and result in a final `undefined` if its not resolved.

Perhaps possible to guarantee non failure from `fromScript` but that would be for another time.

Thanks for feedback 👍🏻